crates/bevy_ecs/src/world/mod.rs

```diff
@@ -444,6 +444,29 @@ impl World {
     /// assert_eq!(world.get::<Position>(entities[0]).unwrap(), &Position { x: 1.0, y: 0.0 });
     /// assert_eq!(world.get::<Position>(entities[1]).unwrap(), &Position { x: 0.0, y: 1.0 });
     /// ```
+    ///
+    /// To iterate over entities in a deterministic order,
+    /// sort the results of the query using the desired component as a key.
```

Comment:
I think that the fact this allocates a `Vec<_>` with all entities is a drawback that should be mentioned